### PR TITLE
Position of the only set bit

### PR DESCRIPTION
**Pull request by RaveenaBhasin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

To find the position of the only set bit in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#6615

### CODE REVIEW

**Comment by RaveenaBhasin** [comment]
@ankitaggarwal23 Please review my PR

**Review by ankitaggarwal23 — COMMENTED** [review]
We don't accept such code. It falls neither as a Data Structure not as an algorithm.

**Comment by RaveenaBhasin** [comment]
@ankitaggarwal23 But there are so many questions on bit manipulation already present so what is the problem with this question?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Add README updates as well. Shift the I/O sample at the bottom of the code along with the Time & Space Complexity.

**Comment by RaveenaBhasin** [comment]
@HarshCasper I'll complete the required changes soon.

**Comment by RaveenaBhasin** [comment]
@HarshCasper Required changes completed. Kindly review my PR.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add a new line at the EOF
- Use proper display message how you are reading and writing outputs
- Add input-output prompts to make the program more interactive.
- comments you add are above the line of code and not alongside it.

**Comment by RaveenaBhasin** [comment]
@ankitaggarwal23 I have added the description, removed extra spaces, and added a new line at the EOF else everything was already present.

File: C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#include<bits/stdc++.h>
+using namespace std;
+int findPosition(int n){
+    int count=0;
+    int position=0;
+    while(n!=0){
+        if(n&1){
+            count++;
+        }
+        n=n>>1;   //Using right shift operator
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp:14]: You need to add comments to your code. Make sure the comments you add are above the line of code and not alongside it.

File: C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    else{
+        return -1;
+    }
+}
+int main(){
+    int n;
+    cin>>n;
+    cout<<findPosition(n)<<endl;
+    return 0;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp:28]: Add input-output prompts to make the program more interactive.

File: C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Example 1:
+4
+Example 2:
+15
+Sample Output
+Example 1:
+3      //4 is represented as "100" in Binary. As we see there's only one set bit and it's in Position 3 
+Example 2:
+-1     //15 is represented as "1111" in Binary. As we see there's four set bits and thus the Output -1
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/Bit--Manipulation/Position_of_the_only_set_bit.cpp:43]:
```
/*

Sample Input/Output:

Input:

Output:


Time complexity:
Space complexity:

*/
```

Use this format

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity. Thank you for your valuable time.😊

**Pull request closed without merging**